date1.java

```diff
@@ -0,0 +1,188 @@
+
+                    if (dueWeeks == 0) {
+                        LocalDateTime temp1 = systemTime.withHour(LstartDate.getHour())
+                                .withMinute(LstartDate.getMinute()).withSecond(LstartDate.getSecond());
+                        if ((temp1.getDayOfWeek().getValue() <= order[dayList.length - 1]))
+                            nextDate = temp1;
+                        else
+                            nextDate = nextDate.with(DayOfWeek.MONDAY).plusWeeks(repeatEvery);
+                    } else {
+                        nextDate = nextDate.plusWeeks(repeatEvery - dueWeeks);
+                        nextDate = nextDate.with(DayOfWeek.MONDAY);
+                        nextDate = nextDate.plusDays(order[0] - 1);
+                    }
+                }
+                
+                
+                while(true)
+                {
+                    if(Arrays.stream(dayList).anyMatch(nextDate.getDayOfWeek().name().substring(0, 3)::equals) && nextDate.isAfter(systemTime))
+                    {
+                        if(LendDate.isAfter(nextDate))
+                            return Date.from(nextDate.atZone(currentZone).toInstant());
+                        else
+                            return null;
+                    }
+
+                    nextDate=nextDate.plusDays(1);
+                }
+            }
+
+
+            case Monthly:
+            {
+                int j=(runOnDay=="last")?0:Integer.parseInt(runOnDay);
+                if(LstartDate.isAfter(systemTime))
+                    nextDate=LstartDate;
+                else
+                    nextDate=systemTime.withHour(LstartDate.getHour()).withMinute(LstartDate.getMinute()).withSecond(LstartDate.getSecond());
+                    
+                LocalDateTime temp=nextDate;
+
+                while(true)
+                {   
+                    //nextDate= (j==0)?nextDate.with(TemporalAdjusters.lastDayOfMonth()) : nextDate.withDayOfMonth(Integer.parseInt(runOnDay));
+                    if(Arrays.stream(monthList).anyMatch(nextDate.getMonth().name().substring(0,3)::equals) && nextDate.isAfter(systemTime) && nextDate.toLocalDate().lengthOfMonth()>=j)
+                    {
+                        nextDate= (j==0)?nextDate.with(TemporalAdjusters.lastDayOfMonth()) : nextDate.withDayOfMonth(Integer.parseInt(runOnDay));
+                    
+                        if(nextDate.isAfter(LstartDate) || nextDate==LstartDate)
+                        {
+                            if(LendDate.isAfter(nextDate))
+                                return Date.from(nextDate.atZone(currentZone).toInstant());
+                            else
+                                return null;
+                        }
+                    }
+
+                    nextDate=nextDate.plusMonths(1);
+                    if(ChronoUnit.MONTHS.between(temp, nextDate)>12)
+                        return null;
```

Comment:
this code has to be cleaned till depth. 
If there is a complex logic then make sure we have proper comments on them.